### ISSUE TITLE
Exception for Non-Reserved Keyword Used as Column Name 

### ISSUE THREAD

**Issue by reata** [reata/sqllineage#183]
```sql
INSERT OVERWRITE TABLE tab1
SELECT CASE WHEN locate('.', a.col1) > 0 THEN substring_index(a.col1, '.', 3) END host
FROM tab2 a
```

Non-reserved keyword: host, desc, summary for example.

**Comment by reata** [comment]
Note TPS-DS query49.sql also suffers from this issue when non-reserve keyword catalog used as CTE name.

**Comment by reata** [comment]
Let's fix #93 first, because in essence these two issues are the same thing.

**Comment by reata** [comment]
It's working now, as we can use sqlfluff sparksql dialect for parsing.
